### PR TITLE
OEL-254 Create Search server and index for oe_showcase

### PR DESCRIPTION
**Pull request by GilNovacomm**

### CODE REVIEW

File: modules/oe_showcase_search/tests/src/Functional/ShowcaseSearchAssets.php [hunk 1]
```diff
@@ -0,0 +1,134 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\Tests\oe_showcase_search\Functional;
+
+use Drupal\Core\Url;
```
Comment (Maxfire) [inline modules/oe_showcase_search/tests/src/Functional/ShowcaseSearchAssets.php]: Unused use statement

File: modules/oe_showcase_search/tests/src/Functional/ShowcaseSearchAssets.php [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+   * {@inheritdoc}
+   */
+  protected static $modules = [
+    'user',
+    'node',
+    'system',
+    'search_api',
+    'search_api_db',
+    'oe_showcase_search',
+    'oe_showcase_default_content'
```
Comment (Maxfire) [inline modules/oe_showcase_search/tests/src/Functional/ShowcaseSearchAssets.php]: A comma should follow the last multiline array item.

File: modules/oe_showcase_search/tests/src/Functional/ShowcaseSearchAssets.php [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+  ];
+
```
Comment (Maxfire) [inline modules/oe_showcase_search/tests/src/Functional/ShowcaseSearchAssets.php]: Whitespace found at end of lines 32, 44, 60, 81

File: modules/oe_showcase_search/tests/src/Functional/ShowcaseSearchAssets.php [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+   */
+  protected function setUp(): void {
+    parent::setUp();
+    $this->adminUser = $this->drupalCreateUser([], NULL, TRUE);
+  }
+
+  /**
+   * Tests whether the default search was correctly installed.
+   */
+  public function testInstallAndDefaultSetupWorking(): void  {
```
Comment (Maxfire) [inline modules/oe_showcase_search/tests/src/Functional/ShowcaseSearchAssets.php]: Expected 1 space before opening brace.

File: modules/oe_showcase_search/tests/src/Functional/ShowcaseSearchAssets.php [hunk 5]
```diff
@@ -0,0 +1,134 @@
...
+
+    // Assure the Showcase index can be used.
+    $index = Index::load('showcase_search_index');
+    $this->assertInstanceOf(Index::class, $index, 'Index can be loaded');
+  }
+
+  /**
+   * Tests that all fields are added to the index, as expected.
+   */
+  public function testFields(): void  {
```
Comment (Maxfire) [inline modules/oe_showcase_search/tests/src/Functional/ShowcaseSearchAssets.php]: Expected 1 space before opening brace.

File: modules/oe_showcase_search/tests/src/Functional/ShowcaseSearchAssets.php [hunk 6]
```diff
@@ -0,0 +1,134 @@
...
+      $field = $fields[$setting['property_path']];
+      $this->assertEquals($setting['label'], $field->getLabel());
+      $this->assertEquals($setting['datasource_id'], $field->getDatasourceId());
+      $this->assertEquals($setting['type'], $field->getType());
+    }
+  }
+
+ /**
+   * Tests available indexed items.
+   */
```
Comment (Maxfire) [inline modules/oe_showcase_search/tests/src/Functional/ShowcaseSearchAssets.php]: the previous comment is not indented correctly.

File: modules/oe_showcase_search/tests/src/Functional/ShowcaseSearchAssets.php [hunk 7]
```diff
@@ -0,0 +1,134 @@
...
+  public function testIndexedContent(): void  {
```
Comment (Maxfire) [inline modules/oe_showcase_search/tests/src/Functional/ShowcaseSearchAssets.php]: Expected 1 space before opening brace.

File: modules/oe_showcase_search/tests/src/Functional/ShowcaseSearchAssets.php [hunk 8]
```diff
@@ -0,0 +1,134 @@
...
+    // Set index
```
Comment (Maxfire) [inline modules/oe_showcase_search/tests/src/Functional/ShowcaseSearchAssets.php]: full stop needed.

**Review by Maxfire — CHANGES_REQUESTED** [review]
Only CS issues I tested the func and checked the code and looks great, thank you Gilmar.

File: behat.yml.dist [hunk 9]
```diff
@@ -2,7 +2,11 @@ default:
   suites:
     default:
       paths:
+<<<<<<< HEAD
+        - %paths.base%/tests/features
+=======
         - "%paths.base%/tests/features"
```
Comment (drishu) [inline behat.yml.dist]: use this one

File: docker-compose.yml [hunk 10]
```diff
@@ -23,7 +23,26 @@ services:
     # ports:
     #   - 3306:3306
 #### Mac users: uncomment the "volumes" key to enable the NFS file sharing. You can find more information about Docker for Mac here: https://github.com/openeuropa/openeuropa/blob/master/docs/starting/tooling.md#using-docker-on-macos
-
+  # If you would like to see what is going on you can run the following on your host:
+  # docker run --rm -p 4444:4444 -p 5900:5900 --network="host" selenium/standalone-chrome-debug:latest
+  # Newer version of this image might run into this issue:
+  # @link https://github.com/elgalu/docker-selenium/issues/20
+  selenium:
```
Comment (drishu) [inline docker-compose.yml]: we need selenium only in behat, not now

File: modules/oe_showcase_search/tests/src/Functional/SearchIndexTest.php [hunk 11]
```diff
@@ -0,0 +1,133 @@
+<?php
```
Comment (drishu) [inline modules/oe_showcase_search/tests/src/Functional/SearchIndexTest.php:1]: not sure we need this test, we should only add behat tests once we have something facing users, asserting user stories, but we can keep it for now.

File: modules/oe_showcase_search/tests/src/Functional/ShowcaseSearchAssets.php [hunk 12]
```diff
@@ -0,0 +1,133 @@
+<?php
```
Comment (drishu) [inline modules/oe_showcase_search/tests/src/Functional/ShowcaseSearchAssets.php]: remove this one, see SearchIndexTest.php

File: phpunit.xml.dist [hunk 13]
```diff
@@ -3,11 +3,21 @@
   <php>
     <ini name="error_reporting" value="32767"/>
     <ini name="memory_limit" value="-1"/>
+<<<<<<< HEAD
     <env name="MINK_DRIVER_ARGS_WEBDRIVER" value='["${selenium.browser}", null, "${selenium.host}:${selenium.port}/wd/hub"]'/>
```
Comment (drishu) [inline phpunit.xml.dist]: not needed

File: phpunit.xml.dist [hunk 14]
```diff
@@ -3,11 +3,21 @@
...
+>>>>>>> 29c4e50a1c960d9dc6b48e002c59e0a9daf02a50
+    <env name="SIMPLETEST_BASE_URL" value="http://web:8080/${drupal.root}"/>
   </php>
   <testsuites>
     <testsuite name="OpenEuropa Showcase">
+<<<<<<< HEAD
       <directory>./tests/</directory>
+      <directory>./modules/oe_showcase_search/tests/</directory>
+=======
+      <directory>./modules/**/tests/</directory>
```
Comment (drishu) [inline phpunit.xml.dist:10]: use this one

**Review by drishu — CHANGES_REQUESTED** [review]
be sure not to commit conflicts, allways revisit the PR and go through the code, fix anything you see

File: modules/oe_showcase_search/tests/src/Functional/ShowcaseSearchAssets.php (hunk 12 above)
Comment (GilNovacomm) [reply modules/oe_showcase_search/tests/src/Functional/ShowcaseSearchAssets.php]: Not sure how this test got duplicated on SearchIndex. I'll remove it.

**Review by drishu — APPROVED** [review]

**Review by escuriola — APPROVED** [review]

**Review by Maxfire — APPROVED** [review]
Blind approved. :+1:

**Pull request merged**